client/js/editmode.js

```diff
@@ -151,12 +151,12 @@ function generateCardDeckWidgets(id, x, y) {
   const types = {};
   const cards = [];
 
-  [ '1J', '2J' ].forEach(c=>types[c] = { image:`/i/cards-default/${c}.svg` });
+  [ {label:'1J', color: "🃏", suit: "T", alternating:"5J", rank: "J1"}, {label:'2J', color: "🃏", suit: "T", alternating:"5J", rank: "J2"}].forEach(c=>types[c.suit+" "+c.label] = { image:`/i/cards-default/${c.label}.svg` , suit:c.suit, suitColor:c.color, suitAlt:c.alternating, rank:c.rank, rankA:c.rank, rankFixed:c.rank+" "+c.suit});
 
-  [ 'C', 'D', 'H', 'S' ].forEach(function(s) {
-    [ 'A', '2', '3', '4', '5', '6', '7', '8', '9', 'T', 'J', 'Q', 'K' ].forEach(function(n) {
-      types[n+s] = { image:`/i/cards-default/${n}${s}.svg` };
-      cards.push({ id:id+n+s, parent:id+'P', deck:id+'D', type:'card', cardType:n+s });
+  [ {label:'C', color: "♣", alternating:"1♣"}, {label:'D', color: "♦", alternating:"4♦"}, {label:'H', color: "♥", alternating:"2♥"}, {label:'S', color: "♠", alternating:"3♠"} ].forEach(function(s) {
+    [ {label:'A', rank: "01", rankA:"5A"}, {label:'2', rank: "02", rankA:"02"},{label:'3', rank: "03", rankA:"03"},{label:'4', rank: "04", rankA:"04"},{label:'5', rank: "05", rankA:"05"},{label:'6', rank: "06", rankA:"06"},{label:'7', rank: "07", rankA:"07"},{label:'8', rank: "08", rankA:"08"},{label:'9', rank: "09", rankA:"09"},{label:'T', rank: "10", rankA:"10"},{label:'J', rank: "2J", rankA:"2J"},{label:'Q', rank: "3Q", rankA:"3Q"},{label:'K', rank: "4K", rankA:"4K"}].forEach(function(n) {
+      types[s.label+" "+n.rank] = { image:`/i/cards-default/${n.label}${s.label}.svg`, suit:s.label, suitColor:s.color, suitAlt:s.alternating, rank:n.rank,rankA:n.rankA, rankFixed:n.rank+" "+s.label};
+      cards.push({ id:id+"_"+n.label+"_"+s.label, parent:id+'P', deck:id+'D', type:'card', cardType:s.label+" "+n.rank });
```

Comment:
A comment on coding style: wouldn't this be easier to maintain if each key-value pair was on its own line?

e.g.:
```javascript
  [ 
    {
      label: '1J', 
      color: "🃏", 
      suit: "T", 
      ...
    }
  ].forEach(...
```

Reply:
Well, intention is not my strength. I just kept the model That was already there.
I can recommit it if you feel like I should.